data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/PitWorker.java

```diff
@@ -102,17 +106,23 @@ public void run() {
                             sourceCoordinator,
                             indexPartition.get());
 
-                    processIndex(indexPartition.get(), acknowledgementSet.getLeft());
+                    openSearchSourcePluginMetrics.getIndexProcessingTimeTimer().recordCallable(() -> {
```

Comment:
Can you use `Timer::record` which takes a `Runnable` instead? This can let you use a lambda and avoid the `return` statement.

```
openSearchSourcePluginMetrics.getIndexProcessingTimeTimer()
  .recordCallable(() -> processIndex(indexPartition.get(), acknowledgementSet.getLeft()));
```

Reply:
Nice will do